### ISSUE TITLE
"Your first 2D game". Improvement suggestions for users who are complete beginners (No background in game dev or programming)

### ISSUE THREAD

**Issue by warriormaster12** [godotengine/godot-docs#9825]
**Your Godot version:** This documentation was used in 4.1 and 4.2

**Issue description:** I'm a volunteer at Helsinki city's youth organization called [GameDev Club](https://nuorten.hel.fi/en/things-to-do-and-places/hobbies/gaming/gamedev-club/) where we've had an uptick of new club members between ages of 15-17 who may/may not have any experience programming, game development experience or even using computers. 

The club operates similarly to a game jam but with a key distinction: sessions are conducted weekly, allowing participants approximately two hours of development time per meeting.

Completely new members (No experience with programming / game development) start with a tutorial project which we have selected to be [Your first 2D game](https://docs.godotengine.org/en/stable/getting_started/first_2d_game/index.html). We have selected this tutorial because it is fairly simple & quick to go through. 

However, we have observed a few challenges associated with this approach. Given that our members progress through the tutorial incrementally each week, there is a tendency for them to forget certain details between sessions. As an example, connecting signals in player character page vs enemy character page.

----
 [Player page](https://docs.godotengine.org/en/stable/getting_started/first_2d_game/03.coding_the_player.html)
<img width="861" alt="image" src="https://github.com/user-attachments/assets/b5ef2a10-5421-4628-826f-85f50a739dae">

----

[Enemy page](https://docs.godotengine.org/en/stable/getting_started/first_2d_game/04.creating_the_enemy.html#enemy-script)
<img width="845" alt="image" src="https://github.com/user-attachments/assets/29f0aff0-8bec-4ea4-9360-be73b0063184">

----

**Improvement suggestions:** 

- Add more images and gifs/videos so that it is easier for readers to follow what's going on. [Game Maker has a good beginners tutorial which utilizes images/gifs a lot](https://gamemaker.io/en/tutorials/heros-trail-dnd-2). This helps people who might have a harder time to digest written content than visual. On top of that, Gifs can effectively illustrate the functionality of features such as "Grouping nodes."

----
<img width="861" alt="image" src="https://github.com/user-attachments/assets/00834878-005b-417d-a786-bbfabe3a2c62">

----

- Incorporate inter-chapter reminders to reinforce procedural concepts, as repetition enhances retention of essential skills. Example signals.
- This requires more work but maybe create tutorial similar to [Game Maker](https://gamemaker.io/en/tutorials/heros-trail-dnd-2). Instead of making a whole game from scratch, readers learn to create feature(s) instead.

**Comment by skyace65** [comment]
In my personal opinion, your suggestions would not work for this section of the documentation.

Explaining to people how to use things, like signals, like it's their first time every time they show up in the tutorial is way too much. The first 2D game tutorial is made with the assumption that you've already completed the other getting started section tutorials (which go over stuff like signals), and we say so on the first page. Technically speaking it already goes over too many things like this is the first time users are being introduced to concepts. We've obviously made some exceptions since we know people skip around, but there's a limit to what we should do.

On the subject of gifs, I read the game maker tutorial you linked to, and if we used them the way game maker does, we wouldn't enhance the documentation at all. They just seem like they're in the game maker documentation as a "Hey, here's something moving, stay interested" thing.

It sounds like you're trying to have everyone with two different learning styles (reading and video) use the exact same learning material. What I would recommend instead is you do one of two things. One, find a video tutorial on youtube and have your students who want a video follow that, I personally would recommend you check out [this video](https://www.youtube.com/watch?v=LOhfqjmasi0) by Brackeys. Or alternatively, record yourself doing our first 2D game tutorial with narration and have your students follow that. OBS is a great free and opens source screen recording program if you need one, and it can capture audio as well, you can find it [here](https://obsproject.com/).

**Comment by warriormaster12** [comment]
> Explaining to people how to use things, like signals, like it's their first time every time they show up in the tutorial is way too much. 

I agree on that though I do think that adding a screenshot of a **Signals** tab right after for instance _"Connect the screen_exited() signal of the VisibleOnScreenNotifier2D node to the Mob and add this code:"_ in enemy script page should be a good enough of a reminder for readers that they shouldn't just write the line in a text editor. 



Maybe changing following instructions:

> Connect the screen_exited() signal of the VisibleOnScreenNotifier2D node to the Mob and add this code:

to this:

> Connect the screen_exited() signal of the VisibleOnScreenNotifier2D node to the Mob. This will add following code to the Mob script:

Could reduce confusion since the tutorial has only asked readers to connect signals via the editor :thinking: .




> It sounds like you're trying to have everyone with two different learning styles (reading and video) use the exact same learning material.

I agree and I shouldn't have necessarily  brought it up as an argument. We only had a couple of cases in which we have indeed suggested a tutorial from some one like Brackeys. 

> Or alternatively, record yourself doing our first 2D game tutorial with narration and have your students follow that. OBS is a great free and opens source screen recording program if you need one, and it can capture audio as well, you can find it [here](https://obsproject.com/).

I'm aware of the program and the idea of recording & narrating the tutorial myself could be something that I could do in the future :grinning:
